cmd/cosign/cli/sign/sign.go

```diff
@@ -152,10 +155,23 @@ func SignCmd(ro *options.RootOptions, ko options.KeyOpts, regOpts options.Regist
 		}
 
 		if digest, ok := ref.(name.Digest); ok && !recursive {
-			se, err := ociremote.SignedEntity(ref, opts...)
-			if err != nil {
-				return fmt.Errorf("accessing image: %w", err)
+			var se oci.SignedEntity
+
+			_, err := remote.Head(digest)
+			var te *transport.Error
+			if errors.As(err, &te) && te.StatusCode == http.StatusNotFound {
+				emptyimg, err := ociempty.SignedImage(ref)
+				if err != nil {
+					return fmt.Errorf("accessing image: %w", err)
+				}
+				se = emptyimg
```

Comment:
Just checking: Is this going to mean that any attachments get hung off the digest of an empty image? Or does `SignedImage(digest)` handle the mapping of digest->attachments, and ignore the digest of its actual image contents?

An e2e test might help here, something that attaches some content to `@sha256:$(head -n /dev/urandom | sha256sum -)` and gets it back by that same random digest.

Reply:
Yes, it will override any attachment images since it assumes that the image and all of the side images (sbom, sig, etc) are absent simply by doing a HEAD call for the image by digest. should it check for the presence of attachments as well to prevent the override?

Reply:
Also, i am not sure where to put the e2e test. i have checked `test/e2e` folder but couldn't find a place where we assert the behavior with an ephemeral registry present. Could you or someone else put me in right direction?

Reply:
> Yes, it will override any attachment images since it assumes that the image and all of the side images (sbom, sig, etc) are absent simply by doing a HEAD call for the image by digest. should it check for the presence of attachments as well to prevent the override?

Yeah I think that's right. I should be able to attach 2+ things to a non-existent image, e.g., a signature and an SBOM, or 2x signatures.

Reply:
in that case `ociempty.SignedImage` is not going to cover. we need a better version of `ociremote.SignedEntity`

Reply:
or adjust `ociremote.SignedEntity` in a way that doesn't fail when the image isn't present on the remote as you originally suggested.